0027-safely-recommend-actions.adoc

```diff
@@ -0,0 +1,173 @@
+
+== Transition Period
+
+There exist a number of factors, that when combined, represent the transition period.
+
+.Considerations
+* Time to Act (`TTA`)
+* Time to Inclusion (`TTI`)
+* Time to Final (`TTF`)
+* HTLC Expiry time
+
+=== Time to Act
+
+For each actor there is a time frame within which they can be expected to act.
+
+Different users will have different response times e.g., a long running daemon is likely to respond to swap transitions more quickly than a user who is manually checking an application before doing each action. We define Time To Act (`TTA`) as the window of time within which the counterparty can expected to take the next action.
+
+=== Time To Inclusion
+
+For each blockchain there exists a time for a transaction to be included in a block.
+
+The likelihood of a transaction being included in a block is dependant on the transaction fee coupled with the current load on the blockchain network i.e., the number of current unconfirmed transactions. An increase in the transaction fee leads to an increase in the probability that the transaction will be included. Inclusion is not guaranteed, therefore the probability is defined to be the percentage chance of a transaction being included in a block within the next `N` blocks.
+
+With a given fee and a given probability tolerance we define Time To Inclusion for a given blockchain as:
+
+----
+TTI = N * <average block time>
+----
+
+=== Time To Final
+
+For each blockchain there exists a time, after a transaction has been included in a block, that the transaction is considered to be final.
+
+Confirmations represent the number of blocks in the block chain that have been accepted by the network since the block that includes the transaction. The number of confirmations a transaction has determines the probability that a blockchain fork will invalidate the transaction. With a given blockchain we define Time To Final as:
+
+----
+TTF = <number of required confirmations> * <average block time>
+----
+
+=== HTLC Expiry Time
+
+COMIT swaps are atomic due to the functionality provide by Hash Time Locked Contracts. Mapped to COMIT swap parlance, a user may not refund until after the expiry time has elapsed.
+
+The atomic guarantees of the COMIT protocols rely on the beta expiry time being _before_ the alpha expiry time, these expiry times are absolute (as opposed to relative) i.e., seconds since epoch.
+
+In order for a swap to be guaranteed atomic the redeem transaction must be included in the blockchain _before_ the expiry time elapses. Therefore the redeem action is subject to some `TTI` calculated for the given blockchain. Repeating because this is important; the expiry time represents a hard limit, the redeem transaction must be included in a block before the expiry elapses for the swap to be atomic.
```

Comment:
If you fancy, you could go more into detail here of why it is not atomic otherwise: The expiry time represents the earliest moment in time where the refund transaction is considered to be valid and can be broadcasted to the network. Henceforth, starting from this point it becomes a race of which transaction is included first. Given that the two transactions are mutually-exclusive, only one of them can make it.

Reply:
I added your words to the doc.